testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/SecurityAuthCommandsTestCase.java

```diff
@@ -0,0 +1,876 @@
+                    e = ex;
+                }
+            }
+        }
+        if (e != null) {
+            throw e;
+        }
+    }
+
+    @AfterClass
+    public static void cleanup() throws Exception {
+        Exception e = null;
+        if (ctx != null) {
+            try {
+                ctx.handle("/subsystem=elytron/filesystem-realm=" + TEST_FS_REALM + ":remove");
+            } catch (Exception ex) {
+                if (e == null) {
+                    e = ex;
+                }
+            } finally {
+                try {
+                    ctx.handle("reload");
+                } finally {
+                    ctx.terminateSession();
+                }
+            }
+        }
+        if (e != null) {
+            throw e;
+        }
+    }
+
+    @Test
+    public void testOOBSASLManagement() throws Exception {
+        ctx.handle("security enable-sasl-management --no-reload");
+        Assert.assertEquals(ElytronUtil.OOTB_MANAGEMENT_SASL_FACTORY, getManagementInterfaceAuthFactory(ctx, null, true));
+        ctx.handle("security disable-sasl-management --no-reload");
+        Assert.assertNull(getManagementInterfaceAuthFactory(ctx, null, true));
+    }
+
+    @Test
+    public void testOOBSASLManagementItf() throws Exception {
+        ctx.handle("security enable-sasl-management --no-reload --management-interface=" + Util.HTTP_INTERFACE);
+        Assert.assertEquals(ElytronUtil.OOTB_MANAGEMENT_SASL_FACTORY, getManagementInterfaceAuthFactory(ctx, Util.HTTP_INTERFACE, true));
+        ctx.handle("security disable-sasl-management --no-reload --management-interface=" + Util.HTTP_INTERFACE);
+        Assert.assertNull(getManagementInterfaceAuthFactory(ctx, Util.HTTP_INTERFACE, true));
+    }
+
+    @Test
+    public void testOOBSASLManagementNative() throws Exception {
+        enableNative(ctx);
+        try {
+            ctx.handle("security enable-sasl-management --no-reload --management-interface=" + Util.NATIVE_INTERFACE);
+            Assert.assertEquals(ElytronUtil.OOTB_MANAGEMENT_SASL_FACTORY, getManagementInterfaceAuthFactory(ctx, Util.NATIVE_INTERFACE, true));
+            ctx.handle("security disable-sasl-management --no-reload --management-interface=" + Util.NATIVE_INTERFACE);
+            Assert.assertNull(getManagementInterfaceAuthFactory(ctx, Util.NATIVE_INTERFACE, true));
+        } finally {
+            disableNative(ctx);
+        }
+    }
+
+    @Test
+    public void testOOBSASLManagement2() throws Exception {
+        // New factory but reuse OOB ManagementRealm properties realm.
+        // side effect is to create a constant realm mapper for ManagementRealm.
+        ctx.handle("security enable-sasl-management --no-reload --mechanism=DIGEST-MD5 "
+                + "--user-properties-file=mgmt-users.properties --group-properties-file=mgmt-groups.properties  "
+                + "--relative-to=jboss.server.config.dir --exposed-realm=ManagementRealm --new-security-domain-name=" + TEST_DOMAIN
```

Comment:
When I change `--exposed-realm` to ManagementRealmExposed I got test failures.

Test fails on
org.junit.ComparisonFailure: expected:<[ManagementRealm]> but was:<[mgmt-users.properties]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.jboss.as.test.integration.management.cli.SecurityAuthCommandsTestCase.testOOBSASLManagement2(SecurityAuthCommandsTestCase.java:411)

[mchoma@localhost standalone]$ git diff
diff --git a/testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/SecurityAuthCommandsTestCase.java b/testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/SecurityAuthCommandsTestCase.java
index 3884e40757..2dc950dab3 100644
--- a/testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/SecurityAuthCommandsTestCase.java
+++ b/testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/SecurityAuthCommandsTestCase.java
@@ -402,12 +402,12 @@ public class SecurityAuthCommandsTestCase {
         // side effect is to create a constant realm mapper for ManagementRealm.
         ctx.handle("security enable-sasl-management --no-reload --mechanism=DIGEST-MD5 "
                 + "--user-properties-file=mgmt-users.properties --group-properties-file=mgmt-groups.properties  "
-                + "--relative-to=jboss.server.config.dir --exposed-realm=ManagementRealm --new-security-domain-name=" + TEST_DOMAIN
+                + "--relative-to=jboss.server.config.dir --exposed-realm=ManagementRealmExposed --new-security-domain-name=" + TEST_DOMAIN
                 + " --new-auth-factory-name=" + TEST_SASL_FACTORY);
         Assert.assertEquals(TEST_SASL_FACTORY, getManagementInterfaceAuthFactory(ctx, null, true));
 
         // Check Realm.
-        Assert.assertEquals("ManagementRealm", getExposedRealmName(true, "DIGEST-MD5"));
+        Assert.assertEquals("ManagementRealmExposed", getExposedRealmName(true, "DIGEST-MD5"));
         Assert.assertEquals("ManagementRealm", getMechanismRealmMapper(true, "DIGEST-MD5"));
         getNames(Util.CONSTANT_REALM_MAPPER).contains("ManagementRealm");

Reply:
This test expects the exposed realm to be the one from the OOB management properties realm.
When you change the exposed realm, then the command doesn't retrieve a properties-file realm that matches it so it creates a new one. This new one being not expected, it is not cleaned at the end of the test and is found (although not expected) when checking the state of the model.